app/models/account_conversation.rb

```diff
@@ -43,22 +43,21 @@ def participant_accounts
   end
 
   class << self
-    def to_a_paginated_by_id(limit, min_id: nil, max_id: nil, since_id: nil, preload_participants: true)
+    def to_a_paginated_by_id(limit, options = {})
       array = begin
-        if min_id
-          paginate_by_min_id(limit, min_id, max_id).reverse
+        if options[:min_id]
+          paginate_by_min_id(limit, options[:min_id], options[:max_id]).reverse
         else
-          paginate_by_max_id(limit, max_id, since_id).to_a
+          paginate_by_max_id(limit, options[:max_id], options[:since_id]).to_a
         end
       end
 
-      if preload_participants
-        participant_ids = array.flat_map(&:participant_account_ids)
-        accounts_by_id = Account.where(id: participant_ids).index_by(&:id)
+      # Preload participants
+      participant_ids = array.flat_map(&:participant_account_ids)
+      accounts_by_id = Account.where(id: participant_ids).index_by(&:id)
```

Comment:
Does this skips the query if `options[:participant_account_ids]` is empty?

Reply:
Yeah, just checked, and Rails is smart enough to not issue a query if `participant_ids` is empty.